src/libstore/build/entry-points.cc

```diff
@@ -58,6 +58,27 @@ BuildResult Store::buildDerivation(const StorePath & drvPath, const BasicDerivat
         result.status = BuildResult::MiscFailure;
         result.errorMsg = e.msg();
     }
+    // XXX: Should use something like
+    // `queryPartialDerivationOutputMap`, but we can't do that
+    // as it assumes a registered derivation
```

Comment:
There is a `DerivationGoal::queryPartialDerivationOutputMap` for this purpose that you might be able to somehow use?

Reply:
Mh I might indeed. Will need to update it a bit to handle non-static output paths, but I think it's a good thing anyways. Thanks :)

Reply:
At a second though it can't, because `queryPartialDerivationOutputMap` only returns the output paths and not the full realisations. So we'll have to update it first.

I've updated the comment to reflect this

Reply:
OK sounds good.